R/epichains.R

```diff
@@ -55,19 +55,21 @@ new_epichains <- function(tree_df,
 #' @author James M. Azam
 #' @export
 epichains <- function(tree_df,
-                           index_cases,
-                           statistic,
-                           offspring_dist,
-                           stat_max,
-                           track_pop) {
+                      index_cases,
+                      statistic,
+                      offspring_dist,
+                      track_pop,
+                      stat_max = Inf) {
   # Check that inputs are well specified
-  checkmate::assert_data_frame(tree_df)
-  checkmate::assert_integerish(index_cases, null.ok = TRUE)
-  checkmate::assert_character(statistic, null.ok = TRUE)
+  checkmate::assert_data_frame(tree_df, min.cols = 3, min.rows = index_cases)
+  checkmate::assert_count(index_cases, positive = TRUE)
+  checkmate::assert_choice(statistic, choices = c("size", "length"))
   .check_offspring_func_valid(offspring_dist)
-  checkmate::assert_logical(track_pop)
-  checkmate::assert_number(stat_max, null.ok = TRUE)
-
+  checkmate::assert_logical(track_pop, len = 1L)
+  checkmate::assert(
+    is.infinite(stat_max),
+    checkmate::check_integerish(stat_max, lower = 1L)
+  )
```

Comment:
Just to check - this syntax is due to the recent discussion about how `check_integerish()` does not allow `Inf`? I liked the functionality of `checkmate::assert()` where the fail conditions are neatly displayed.

The message is a little confusing though, but that seems like a {checkmate} issue: "one of the following must apply" suggests that the expectation is that `stat_max` _cannot_ be infinite, and _cannot_ be positive - both these are contradicted by the checking code displayed, so a reasonable user would be able to figure out the requirements. But I'm wondering whether there isn't a nice fix? Not a real issue if not.
```
Error: Assertion failed. One of the following must apply:
 * is.infinite(stat_max): FALSE
 * checkmate::check_integerish(stat_max): Element 1 is not >= 1
```

Reply:
> Just to check - this syntax is due to the recent discussion about how check_integerish() does not allow Inf?

Yes, I'm faced with a check where you want something to be `Inf` or `integer`(or easily coercible to integer).

> The message is a little confusing though, but that seems like a {checkmate} issue: "one of the following must apply" suggests that the expectation is that stat_max cannot be infinite, and cannot be positive - both these are contradicted by the checking code displayed, so a reasonable user would be able to figure out the requirements. But I'm wondering whether there isn't a nice fix? Not a real issue if not.

It is quite confusing at first glance. Basically, it throws the error message or FALSE due to failure. I might be gather throw custom/more readable error messages using `checkmate:: reportAssertions()`. I'll look into it and see how I can improve the error messaging here in future iterations.